src/storage/src/hummock/shared_buffer/shared_buffer_batch.rs

```diff
@@ -78,6 +120,7 @@ pub(crate) struct SharedBufferKeyEntry {
 pub(crate) struct SharedBufferBatchInner {
     entries: Vec<SharedBufferKeyEntry>,
     new_values: Vec<VersionedSharedBufferValue>,
+    old_values: Option<Vec<Bytes>>,
```

Comment:
It seems that we assume `new_values` and `old_values` have the same length. We should add an assertion for it. Also, the `value_offset` documentation needs to be updated accordingly in L112.